util/cache/inmemory.go

```diff
@@ -29,6 +29,21 @@ func (i *InMemoryCache) Set(item *Item) error {
 	return nil
 }
 
+func (i *InMemoryCache) HasSame(key string, obj interface{}) bool {
+	var buf bytes.Buffer
+	err := gob.NewEncoder(&buf).Encode(obj)
+	if err != nil {
+		return false
+	}
```

Comment:
Could it be that we'll silently drop some real errors here? In which cases does `gob.NewEncoder()` fail?

Reply:
Agree, change method so it returns an error instead of silently dropping it